test/integration/analytical.cpp

```diff
@@ -0,0 +1,1539 @@
+
+  // B->C reaction rate
+  a_ = 2.0e-22 * std::exp(2) * 2.45e19;
+  b_ = 0.43 * std::pow((293.0 / 298.0), -8.0);
+  z = a_ / (1.0 + a_ / b_) * std::pow(0.41, 1.0 / (1.0 + std::pow(std::log10(a_ / b_), 2))) * (1.0 - 1.0e-3) / 1.0e-3;
+  a_ = 2.0e-22 * std::exp(2) * air_dens_n_cm3;
+  b_ = 0.43 * std::pow((temperature / 298.0), -8.0);
+  A = a_ / (1.0 + a_ / b_) * std::pow(0.41, 1.0 / (1.0 + std::pow(std::log10(a_ / b_), 2)));
+
+  double k2 = 1.2 * std::exp(-204.3 / temperature) * (A / (z + A));
+
+  double time_step = 1.0;
+  micm::State<micm::Matrix> state = solver.GetState();
+
+  std::vector<std::vector<double>> model_concentrations(nsteps, std::vector<double>(3));
+  std::vector<std::vector<double>> analytical_concentrations(nsteps, std::vector<double>(3));
+
+  model_concentrations[0] = { 1, 0, 0 };
+  analytical_concentrations[0] = { 1, 0, 0 };
+
+  state.variables_[0] = model_concentrations[0];
+  state.conditions_[0].temperature_ = temperature;
+  state.conditions_[0].pressure_ = pressure;
+  state.conditions_[0].air_density_ = air_density;
+
+  size_t idx_A = 0, idx_B = 1, idx_C = 2;
+
+  for (size_t i_time = 1; i_time < nsteps; ++i_time)
+  {
+    // Model results
+    auto result = solver.Solve(time_step, state);
+    EXPECT_EQ(result.state_, (micm::RosenbrockSolver<micm::Matrix, SparseMatrixTest>::SolverState::Converged));
+    EXPECT_EQ(k1, state.rate_constants_.AsVector()[0]);
+    EXPECT_EQ(k2, state.rate_constants_.AsVector()[1]);
+    model_concentrations[i_time] = result.result_.AsVector();
+    state.variables_[0] = result.result_.AsVector();
+
+    // Analytical results
+    double time = i_time * time_step;
+
+    double initial_A = analytical_concentrations[0][idx_A];
+    analytical_concentrations[i_time][idx_A] = initial_A * std::exp(-(k1)*time);
+    analytical_concentrations[i_time][idx_B] = initial_A * (k1 / (k2 - k1)) * (std::exp(-k1 * time) - std::exp(-k2 * time));
+
+    analytical_concentrations[i_time][idx_C] =
+        initial_A * (1.0 + (k1 * std::exp(-k2 * time) - k2 * std::exp(-k1 * time)) / (k2 - k1));
+  }
+
+  std::vector<std::string> header = { "time", "A", "B", "C" };
+  writeCSV("analytical_concentrations.csv", header, analytical_concentrations);
+  writeCSV("model_concentrations.csv", header, model_concentrations);
+
+  auto map = state.variable_map_;
+
+  size_t _a = map.at("A");
+  size_t _b = map.at("B");
+  size_t _c = map.at("C");
+
+  for (size_t i = 0; i < model_concentrations.size(); ++i)
+  {
+    EXPECT_NEAR(model_concentrations[i][_a], analytical_concentrations[i][0], 1e-3)
+        << "Arrays differ at index (" << i << ", " << 0 << ")";
+    EXPECT_NEAR(model_concentrations[i][_b], analytical_concentrations[i][1], 1e-3)
+        << "Arrays differ at index (" << i << ", " << 1 << ")";
+    EXPECT_NEAR(model_concentrations[i][_c], analytical_concentrations[i][2], 1e-3)
+        << "Arrays differ at index (" << i << ", " << 2 << ")";
```

Comment:
@mattldawson for some reason I had to drop the tolerance for this reaction to 1e-3. I suspect it's another floating point thing, but I'm not sure if it indicates something else that may be wrong